test/test_jit_cuda_fuser.py

```diff
@@ -2546,6 +2546,23 @@ def t(x: torch.Tensor, weight: torch.Tensor, bias: torch.Tensor):
         # have been optimized away
         self.assertGraphContainsExactly(t_jit.graph_for(x, weight, bias), FUSION_GUARD, 1)
 
+    @unittest.skipIf(not RUN_NVFUSER, "requires CUDA")
+    @unittest.skipIf(GRAPH_EXECUTOR != ProfilingMode.PROFILING,
+                     "Requires fusion optimization pass to be effective")
+    def test_linear_symbolic_shapes(self):
+        def fn(x: int):
+            y = torch.zeros((x, x + 2)).cuda()
+            for i in range(2):
+                inp = torch.rand((x, x + i)).cuda()
+                weight = torch.rand((x + 2, x + i)).cuda()
+                bias = torch.rand((x, x + 2)).cuda()
+                y += torch.sin(torch.nn.functional.linear(inp, weight, bias))
```

Comment:
The for-loop would run twice during profile runs and see different shapes, which would resolve with symbolic shape in PE.

Our consumption of profiling information is based on everything being static to keep it simple.